lib/defopt.py

```diff
@@ -563,16 +586,28 @@ def _populate_parser(func, parser, parsers, short, cli_options,
             kwargs['metavar'] = '{' + ','.join(type_.__members__) + '}'
         elif _ti_get_origin(type_) is tuple:
             member_types = _ti_get_args(type_)
+            num_members = len(member_types)
             # Variable-length tuples of homogenous type are specified like
             # Tuple[int, ...]
-            if len(member_types) == 2 and member_types[1] is Ellipsis:
-                kwargs['nargs'] = "*"
-                kwargs['action'] = _make_store_tuple_action_class(
-                    tuple, member_types, parsers, is_variable_length=True)
+            if num_members == 2 and member_types[1] is Ellipsis:
+                kwargs["nargs"] = "*"
+                kwargs["action"] = _make_store_tuple_action_class(
+                    tuple, member_types, parsers, is_variable_length=True
+                )
+            elif (
+                num_members > 1
+                and type(None) in union_args
+                and parsers.get(type(None))
+            ):
+                raise ValueError(
+                    "Optional tuples of length greater than one are not "
+                    "supported with use of a NoneType parser due to ambiguity."
+                )
```

Comment:
I guess we should also disable support for 1-element tuples for now, per https://github.com/anntzer/defopt/issues/115#issuecomment-1159549912?

Reply:
done